src/lex.ll

```diff
@@ -706,64 +714,186 @@ lCComment(SourcePos *pos) {
     Error(*pos, "unterminated comment");
 }
 
+static void lNextValidChar(SourcePos *pos, char const*&currChar) {
+    while (*currChar == ' ') {
+        ++pos->last_column;
+        currChar++;
+    }
+}
+
+/** Handle pragma directive to unroll loops.
+*/
+static void lPragmaUnroll(YYSTYPE *yylval, SourcePos *pos, std::string fromUserReq, bool isNounroll) {
+
+    const char *currChar = fromUserReq.data();
+    yylval->pragmaAttributes = new PragmaAttributes();
+    yylval->pragmaAttributes->aType = PragmaAttributes::AttributeType::pragmaloop;
+    int count = -1;
+
+    lNextValidChar(pos, currChar);
+
+    if (isNounroll) {
+        if (*currChar == '\n') {
+            yylval->pragmaAttributes->unrollType = Globals::pragmaUnrollType::nounroll;
+            pos->last_column = 1;
+            pos->last_line++;
+            return;
+        }
+        pos->last_column = 1;
+        pos->last_line++;
+        Error(*pos, "extra tokens at end of '#pragma nounroll'.");
+        return;
+
+    }
+
+    if (*currChar == '\n') {
+        yylval->pragmaAttributes->unrollType = Globals::pragmaUnrollType::unroll;
+        pos->last_column = 1;
+        pos->last_line++;
+        return;
+    }
+
+    bool popPar = false;
+    if (*currChar == '(') {
+        popPar = true;
+        currChar++;
+        ++pos->last_column;
+    }
+
+    char *endPtr = NULL;
+#if defined(ISPC_HOST_IS_WINDOWS) && !defined(__MINGW32__)
```

Comment:
``std::stoi()`` seem to be better alternative here, at least it's cross-platform.
It also doesn't return 0 on failure, but instead indicates it through not changed pointer (at least documentation says that).

Reply:
std::stoi throws an exception for characters...something like 'aa'. We can use a try, catch block I suppose. Plus the code I used is the same as we use for identifying integers elsewhere in the lexer. If we plan to change, maybe we should take care of both together?

Reply:
If this is already the code used in this file, then ok.